src/lib.rs

```diff
@@ -82,36 +85,91 @@
+//!      .build();
+//!
+//! let token = sign_access_token(&mut backend, &sign_key, claims, &None, Some(unprotected_header), None)?;
+//! assert!(verify_access_token(&mut backend, &sign_key, &token, &None).is_ok());
+//! # Ok::<(), AccessTokenError<<OpensslContext as CryptoBackend>::Error>>(())
+//! ```
+//!
 //! # Provided Data Models
 //!
 //! ## Token Endpoint
 //! The most commonly used models will probably be the token endpoint's
-//! [`AccessTokenRequest`] and [`AccessTokenResponse`] described in
-//! [section 5.8 of RFC 9200](https://www.rfc-editor.org/rfc/rfc9200#section-5.8).
+//! [`AccessTokenRequest`](https://docs.rs/dcaf/latest/dcaf/struct.AccessTokenRequest.html) and
+//! [`AccessTokenResponse`](https://docs.rs/dcaf/latest/dcaf/struct.AccessTokenResponse.html)
+//! described in [section 5.8 of RFC 9200](https://www.rfc-editor.org/rfc/rfc9200#section-5.8).
 //! In case of an error, an [`ErrorResponse`] should be used.
 //!
 //! After an initial Unauthorized Resource Request Message, an
-//! [`AuthServerRequestCreationHint`]
+//! [`AuthServerRequestCreationHint`](https://docs.rs/dcaf/latest/dcaf/struct.AuthServerRequestCreationHint.html)
 //! can be used to provide additional information to the client, as described in
 //! [section 5.3 of RFC 9200](https://www.rfc-editor.org/rfc/rfc9200#section-5.3).
 //!
 //! ## Common Data Types
 //! Some types used across multiple scenarios include:
-//! - [`Scope`] (as described in
+//! - [`Scope`](https://docs.rs/dcaf/latest/dcaf/enum.Scope.html) (as described in
 //!   [section 5.8.1 of RFC 9200](https://www.rfc-editor.org/rfc/rfc9200#section-5.8.1)),
-//!   either as a [`TextEncodedScope`], a [`BinaryEncodedScope`] or an [`AifEncodedScope`].
-//! - [`ProofOfPossessionKey`] as specified in
-//!   [section 3.1 of RFC 8747](https://www.rfc-editor.org/rfc/rfc8747#section-3.1).
+//!   either as a [`TextEncodedScope`](https://docs.rs/dcaf/latest/dcaf/struct.TextEncodedScope.html),
+//!   a [`BinaryEncodedScope`](https://docs.rs/dcaf/latest/dcaf/struct.BinaryEncodedScope.html) or
+//!   an [`AifEncodedScope`](https://docs.rs/dcaf/latest/dcaf/struct.AifEncodedScope.html).
+//! - [`ProofOfPossessionKey`](https://docs.rs/dcaf/latest/dcaf/enum.ProofOfPossessionKey.html) as
+//!   specified in [section 3.1 of RFC 8747](https://www.rfc-editor.org/rfc/rfc8747#section-3.1).
 //!   For example, this will be used in the access token's `cnf` claim.
 //! - While not really a data type, various constants representing values used in ACE-OAuth
-//!   are provided in the [`constants`] module.
+//!   are provided in the [`constants`](https://docs.rs/dcaf/latest/dcaf/constants/index.html) module.
 //!
 //! # Token handling
 //!
 //! This crate also provides some functionality regarding the encoding and decoding of access
-//! tokens, especially of CBOR Web Tokens.
+//! tokens, especially of CBOR Web Tokens (CWTs), which are based on the COSE specification
+//! (RFC 9052).
```

Comment:
Wouldn't it make more sense to reference RFC 8392 here if we're talking about CWTs?

Reply:
```suggestion
//! tokens, especially of CBOR Web Tokens (CWTs, [RFC 8392](https://datatracker.ietf.org/doc/html/rfc8392)), 
//! which are based on the COSE specification ([RFC 9052](https://datatracker.ietf.org/doc/html/rfc9052)).
```